pipeline_tools/http_requests.py

```diff
@@ -238,9 +245,10 @@ def _http_request(self, *args, **kwargs):
     @staticmethod
     def _get_method(http_method):
         session = requests.Session()
-        # The default max_redirects is 30. We need to boost it because DSS sometimes redirects us more than 30 times,
-        # which causes an error if we don't allow for more.
-        session.max_redirects = 100
+        retry_policy = RetryPolicy()
```

Comment:
Would be helpful to have a comment about why we need this adapter / retry policy stuff.